src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c

```diff
@@ -427,13 +437,31 @@ zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, i
 	kfree(cmd);
 }
 
+static inline void
+zcu_xgq_cmd_complete_with_status(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret, int status)
+{
+	struct xgq_com_queue_entry r;
+	struct zocl_cu_xgq *zcu_xgq = platform_get_drvdata(pdev);
+
+	if (likely(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq))) {
+		zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, NULL);
+	} else if (status == KDS_SKCRASHED) {
+		init_resp_abort(&r, cmd->cid, ret, status);
+		zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, &r);
+	} else {
+		init_resp(&r, cmd->cid, ret);
+		zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, &r);
+	}
```

Comment:
I would recommend moving `zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, &r); `out of the if-else if-else control.
Something like:```

struct xgq_com_queue_entry r;
struct xgq_com_queue_entry *rptr = &r;

if (likely(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq))) 
  rptr = NULL;
else if  (status == KDS_SKCRASHED) 
  init_resp_abort(rptr, cmd->cid, ret, status);
else
   init_resp(rptr, cmd->cid, ret);
zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, rptr);
```